cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -337,6 +339,41 @@ struct ISOWeek {
   Localizer localizer_;
 };
 
+template <typename Duration, typename Localizer>
+struct Week {
+  explicit Week(const DayOfWeekOptions* options, Localizer&& localizer)
+      : localizer_(std::move(localizer)), count_offset_(options->one_based_numbering) {
+    if (options->week_start == 7) {
+      start_week_ = sun;
+      mid_week_ = wed;
+
+    } else {
+      start_week_ = mon;
+      mid_week_ = thu;
+    }
```

Comment:
Do we intend to only support these two cases? If so we should validate it somewhere and make sure it's described in the docstring.